### PR TITLE
`follow_redirects`: Add `:max_redirects` option

### LINKED ISSUE #64: `follow_redirects`: Add `:max_redirects` option

### PR DESCRIPTION
**Pull request by aleDsz**
- Closes #64

### CODE REVIEW

File: lib/req/steps.ex [hunk 1]
```diff
@@ -752,29 +756,41 @@ defmodule Req.Steps do
...
+      request =
+        request
+        |> build_redirect_request(response)
+        |> Req.Request.put_private(:req_redirect_count, redirect_count + 1)
 
-    {_, result} = Req.Request.run(request)
-    {Req.Request.halt(request), result}
+      {_, result} = Req.Request.run(request)
+      {request, result}
+    else
```
Comment (wojtekmach) [inline lib/req/steps.ex:771]: let's follow what curl does: if we have exceeded max redirects, let's raise an error: `"too many redirects (#{max_redirects})"`

File: test/req/steps_test.exs [hunk 2]
```diff
@@ -592,6 +592,30 @@ defmodule Req.StepsTest do
...
+        assert_raise RuntimeError, "too many redirects (3)", fn ->
+          Req.get!(c.url, max_redirects: 3)
+        end
+      end)
+
+    assert_receive :ping, 1000
+    assert_receive :ping, 1000
+    assert_receive :ping, 1000
+    assert_receive :ping, 1000
+    refute_receive :ping, 1000
```
Comment (wojtekmach) [inline test/req/steps_test.exs:614]:
```suggestion
    assert_receive :ping
    assert_receive :ping
    assert_receive :ping
    assert_receive :ping
    refute_receive _
```

File: lib/req/steps.ex [hunk 3]
```diff
@@ -736,12 +736,16 @@ defmodule Req.Steps do
...
 
+  If we reach the maxium number of redirects, we will raise an exception.
+
   ## Request Options
 
     * `:location_trusted` - by default, authorization credentials are only sent
       on redirects to the same host. If `:location_trusted` is set to `true`, credentials
       will be sent to any host.
 
+    * `:max_redirects` - the maximum number of redirects, defaults to `10`.
```
Comment (wojtekmach) [inline lib/req/steps.ex:747]: please mention this on Req.request/1 too. After that we're ready to go, thanks!

Comment (aleDsz) [reply lib/req/steps.ex:747]: Shout we add `location_trusted` too?

Comment (wojtekmach) [reply lib/req/steps.ex:747]: Yup, good catch.

File: lib/req.ex [hunk 4]
```diff
@@ -369,6 +369,14 @@ defmodule Req do
     * `:params` - if set, appends parameters to the request query string (via
       [`put_params`](`Req.Steps.put_params/1`) step).
 
+  Redirect options:
+
+    * `:location_trusted` - by default, authorization credentials are only sent
+      on redirects to the same host. If `:location_trusted` is set to `true`, credentials
+      will be sent to any host.
+
+    * `:max_redirects` - the maximum number of redirects, defaults to `10`.
```
Comment (wojtekmach) [inline lib/req.ex:378]: please move these to the "Redirect options" section below.

**Comment by aleDsz** [comment]
I've force pushed because the retry options got conflicted with redirect options on the `Req.request/1` docs

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**